packages/table/src/Row.tsx

```diff
@@ -0,0 +1,323 @@
+const idAllocator = IdAllocator.create('text-input');
+
+interface RowProps extends React.ComponentPropsWithoutRef<'tr'> {
+  expanded?: boolean;
+  disabled?: boolean;
+  indentLevel?: number;
+  isAnyAncestorCollapsed?: boolean;
+}
+
+const Row = React.forwardRef(
+  (
+    {
+      expanded = false,
+      disabled = false,
+      indentLevel = 0,
+      isAnyAncestorCollapsed: isAnyAncestorCollapsedProp,
+      children,
+      className,
+      ...rest
+    }: RowProps,
+    ref: React.Ref<any>,
+  ) => {
+    const {
+      state: { data, columnInfo, hasNestedRows, hasRowSpan },
+      dispatch: tableDispatch,
+    } = useTableContext();
+
+    const indexRef = useRef(idAllocator.generate());
+    const [isExpanded, setIsExpanded] = useState(expanded);
+    const nodeRef = useRef(null);
+
+    useEffect(() => {
+      let shouldDispatchHasNestedRows = false;
+      let shouldDispatchHasRowSpan = false;
+
+      React.Children.forEach(children, child => {
+        if (
+          isComponentType(child, 'Row') &&
+          !shouldDispatchHasNestedRows &&
+          !hasNestedRows
+        ) {
+          shouldDispatchHasNestedRows = true;
+        }
+
+        if (
+          isComponentType(child, 'Cell') &&
+          child.props.rowSpan > 1 &&
+          !hasRowSpan &&
+          !shouldDispatchHasRowSpan
+        ) {
+          shouldDispatchHasRowSpan = true;
+        }
+      });
+
+      if (shouldDispatchHasNestedRows) {
+        tableDispatch({
+          type: TableTypes.SetHasNestedRows,
+          payload: true,
+        });
+      }
+
+      if (shouldDispatchHasRowSpan) {
+        tableDispatch({
+          type: TableTypes.SetHasRowSpan,
+          payload: true,
+        });
+      }
+    }, [children, hasNestedRows, hasRowSpan, tableDispatch]);
+
+    // Iterating over children twice because generated memoized values have different dependants
+    const renderedChildren = React.useMemo(() => {
+      const Icon = isExpanded ? ChevronDownIcon : ChevronRightIcon;
+      const chevronButton = (
+        <IconButton
+          onClick={() => setIsExpanded(curr => !curr)}
+          aria-label="chevron"
+          className={iconButtonMargin}
+        >
+          <Icon aria-label="chevron" color={uiColors.gray.dark2} />
+        </IconButton>
+      );
+      const renderedChildren: Array<React.ReactElement> = [];
+      let hasSeenRow = false;
+
+      React.Children.forEach(children, (child, index) => {
+        if (isComponentType(child, 'Cell')) {
+          if (!child.props.children) {
+            return null;
+          }
+
+          renderedChildren.push(
+            React.cloneElement(child, {
+              children: (
+                <span className={truncation}>{child.props.children}</span>
+              ),
+              key: `${indexRef.current}-${index}`,
+              disabled: child.props.disabled || disabled,
+            }),
+          );
+        } else {
+          if (isComponentType(child, 'Row')) {
```

Comment:
Can merge these two lines.

Reply:
Does htis even need to be an else if? Can it be its own if?

Reply:
You're correct